scripts/dts/gen_defines.py

```diff
@@ -132,6 +132,9 @@ def main():
                 out_dt_define(f"{node.z_path_id}_CHILD_IDX",
                               node.parent.child_index(node))
 
+                out_comment(f"Node's number of children:")
+                out_dt_define(f"{node.z_path_id}_CHILD_NUM", len(node.children))
```

Comment:
This should not be part of `if node.parent is not None:`, and could probably move to `write_children`

Reply:
> This should not be part of `if node.parent is not None:`, and could probably move to `write_children`

May I ask why? All the other things like CHILD_INDEX are under that if statement.

Reply:
Because `CHILD_IDX` is the current node's index relative to it's parent. The current node's number of children does not depend on having a parent.

Reply:
> Because `CHILD_IDX` is the current node's index relative to it's parent. The current node's number of children does not depend on having a parent.

The difference is really whether we need a child number for the top node. Adding that outside of the if statement creates a DT_N_CHILD_NUM for the top node. Is that expected?

Reply:
IMO yes, same as `write_children` is called for the top node.

Reply:
Done, thanks. Don't approve yet, lets see what others thinks. Plus I will probably need to rebase.